tests/datasets/test_geo.py

```diff
@@ -63,6 +63,14 @@ def __len__(self) -> int:
         return 2
 
 
+class MockRasterDataset(RasterDataset):
```

Comment:
Let's remove this and instead test to ensure that an error is raised: https://docs.pytest.org/en/latest/how-to/assert.html#assertions-about-expected-exceptions

Reply:
What I want to test here is the behaviour of the `files`-property: that virtual files are added to the list even though they dont exists locally. 

But that cannot be tested without calling init, which raises. If I change the test to check that it raises, it then communicates to other developers that it is testing something different.

What do you think?